TrackballControls: Fixed bad zoom performance

boulabiar opened the pull request:
This is only a fix for chrome #14013

boulabiar committed:
Fix #14013

This is only a fix for chrome

Mugen87 commented on examples/js/controls/TrackballControls.js:
```diff
@@ -12,6 +12,7 @@ THREE.TrackballControls = function ( object, domElement ) {
 
 	this.object = object;
 	this.domElement = ( domElement !== undefined ) ? domElement : document;
+        document.body.style.touchAction = "none";
```
Please use hard tabs instead of soft tabs. Besides, I would write it like this:
```js
document.body.style.touchAction = `none`; // see #14013
```

boulabiar committed:
This is only a fix for chrome touch policy #14013

Chrome changed its policy of handling touch input. In order to restore older behavior and performance, this line is needed.

Mugen87 commented:
Um, something strikes me. This fix works for fullscreen applications. But what happens when the canvas is embedded in a web site surrounded by ordinary web content? Setting `document.body.style.touchAction` to `none` would affect the entire page which could be problematic. Maybe the following is more robust:
```js
if ( this.domElement instanceof HTMLElement ) {

     this.domElement.style.touchAction = 'none';

} else {

    document.body.style.touchAction = 'none';

}
```

boulabiar commented:
You are absolutely right. My bad why I didn't thought of this.

boulabiar committed:
Apply change only to the current element #14013 

Only add the behavior change to the canvas when embedded in the page with other elements. Otherwise to all the body element.

mrdoob commented:
This looks like should be a bug report for Chrome instead... 🤔

boulabiar commented:
I reported the bug to chrome
https://bugs.chromium.org/p/chromium/issues/detail?id=841282

But this is intentional! They want it like that.. It's their new policy and they don't want to change it.
Meanwhile, three.js is still getting this issue in that example. (everybody using chrome mobile has it, tested it on 5 phones already)

mrdoob commented:
So the fact that this is not an issue on with `OrbitControls` is because we're calling `event.preventDefault()` in the `touchstart` event. 

https://github.com/mrdoob/three.js/blob/dev/examples/js/controls/OrbitControls.js#L813

Do you mind updating the PR to do that instead? Seems simpler than messing with css styles.

mrdoob commented:
I see... Well, for the time being I'm going to add these just to match the other controls. I'll investigate the issue more in depth next week.

boulabiar commented:
this is the video of the dev branch showing the same problem.
https://www.youtube.com/watch?v=qQov0Eei_uU

I've already reported this here
https://github.com/mrdoob/three.js/pull/14622#issuecomment-410162595

Now with this commit we have a wrong fps indicator.
